lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.py

```diff
@@ -36,7 +37,10 @@ def defaultVariantName():
     return kDefaultVariantName
 
 def createPrimAndAttributes(stage, primPath, mayaReferencePath, mayaNamespace, mayaAutoEdit):
-    prim = stage.DefinePrim(primPath.path, 'MayaReference')
+    try:
+        prim = stage.DefinePrim(primPath.path, 'MayaReference')
+    except (Tf.ErrorException):
+        return Usd.Prim()
```

Comment:
The only way I figured out to get DefinePrim to fail was to set disable permission to edit on the edit target layer. This causes Usd to throw so we catch that here.